x-pack/libbeat/docs/aws-credentials-config.asciidoc

```diff
@@ -0,0 +1,94 @@
+[float]
+=== AWS Credentials Configuration
+To configure AWS credentials, there are two different ways supported:
+
+[float]
+==== Supported Formats
+* Use `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and/or `AWS_SESSION_TOKEN`
+
+Users can either put the credentials into metricbeat module configuration or use
+environment variable `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and/or
+`AWS_SESSION_TOKEN` instead.
+
+[source,yaml]
+----
+metricbeat.modules:
+- module: aws
+  period: 300s
+  metricsets:
+    - ec2
+  access_key_id: '<access_key_id>'
+  secret_access_key: '<secret_access_key>'
+  session_token: '<session_token>'
+----
+
+or
+
+[source,yaml]
+----
+metricbeat.modules:
+- module: aws
+  period: 300s
+  metricsets:
+    - ec2
+  access_key_id: '${AWS_ACCESS_KEY_ID:""}'
+  secret_access_key: '${AWS_SECRET_ACCESS_KEY:""}'
+  session_token: '${AWS_SESSION_TOKEN:""}'
+----
+
+* Use shared AWS credentials file
+
+[source,yaml]
+----
+metricbeat.modules:
+- module: aws
+  period: 300s
+  metricsets:
+    - ec2
+  credential_profile_name: test-mb
```

Comment:
I think we should add a comment here indicating that this is optional, and if omitted the default profile will be used.